docs/modules/release-notes/pages/0.27.adoc

```diff
@@ -19,34 +21,520 @@ To get started, follow xref:pkl-cli:index.adoc#installation[Installation].#
+
+[source,diff]
+----
+-ipAddresses: Listing<String>(toList().contains("127.0.0.1"))
++ipAddresses: Listing<String>(contains("127.0.0.1"))
+----
+
+Additionally, a new method is added to `String`, called {uri-stdlib-String}#splitLimit()[`String.splitLimit()`].
+
+==== Additions to `pkl:EvaluatorSettings`
+
+New properties have been added to `pkl:EvaluatorSettings` (https://github.com/apple/pkl/pull/660[#660], https://github.com/apple/pkl/pull/746[#746]).
+
+These are:
+
+* {uri-stdlib-evaluatorSettingsModule}/#color[`color`]
+* {uri-stdlib-evaluatorSettingsModule}/#externalModuleReaders[`externalModuleReaders`]
+* {uri-stdlib-evaluatorSettingsModule}/#externalResourceReaders[`externalResourceReaders`]
+
+==== `String` to `Number` conversion improvements.
+
+The `String` to `Number` converter methods, for example,  {uri-stdlib-StringToInt}[`String.toInt()`], can now handle underscore separators (https://github.com/apple/pkl/pull/578[#578], https://github.com/apple/pkl/pull/580[#580]).
+
+This better aligns with the source code representation of file:///Users/danielchao/code/apple/pkl-lang.org/build/local/main/current/language-reference/index.html#integers[number literals].
+
+[source,pkl]
+----
+myNum = "1_000".toInt() // <1>
+----
+<1> Result: `1000`
+
+==== New module: `pkl:analyze`
+
+As part of <<import-analysis>>, a new standard library module is added called `pkl:analyze` (https://github.com/apple/pkl/pull/695[#695]).
+
+This module provides an API for computing the total import graph given a set of input modules.
+This API treats the inputs as entrypoints, and produces a graph representing the entire dependency tree, including transitive imports.
+
+Example:
+[source,pkl]
+----
+import "pkl:analyze"
+
+importGraph: analyze.ImportGraph = analyze.importGraph(Set("file:///path/to/my/module.pkl"))
+----
+
+For details, see the {uri-stdlib-analyzeModule}[documentation].
+
+=== Annotations on `PklProject` added to packages
+
+When creating packages, some annotations on a `PklProject` will affect how pkldoc generates documentation.
+
+* `@Unlisted`: Omit the package from publishing documentation.
+* `@Deprecated`: Add information on the package to show that it is deprecated. 
+
+Example:
+
+.PklProject
+[source,pkl]
+----
+@Deprecated
+amends "pkl:Project"
+----
+
+=== Test report improvements
+
+The report generated from `pkl test` has been overhauled and improved (https://github.com/apple/pkl/pull/498[#498], https://github.com/apple/pkl/pull/628[#682], https://github.com/apple/pkl/pull/738[#738], https://github.com/apple/pkl/pull/771[#771]).
+
+* Tests are grouped into either the `facts` or `examples` section.
+* A summary line is added, describing how many tests have passed and failed, and how many assertions have passed and failed.
+* Test results are colored.
+* The ✅ and ❌ emojis are replaced with ✔ and ✘.
+* Thrown errors are reported within the test.
+
+Here is a sneak peek of the new test result output.
+
+image::test_sample.png[]
```

Comment:
btw, should we include light-themed screenshots? I asked, because running them there surfaced quite a few styling bugs when I was working on the error colours.